crates/bevy_asset/src/path.rs

```diff
@@ -658,56 +715,49 @@ impl<'de> Visitor<'de> for AssetPathVisitor {
 
 /// Normalizes the path by collapsing all occurrences of '.' and '..' dot-segments where possible
 /// as per [RFC 1808](https://datatracker.ietf.org/doc/html/rfc1808)
-pub(crate) fn normalize_path(path: &Path) -> PathBuf {
-    let mut result_path = PathBuf::new();
-    for elt in path.iter() {
+pub(crate) fn normalize_path<'b>(path: &'b [&'b str]) -> Vec<&'b str> {
```

Comment:
I might be missing something... It seems like we don't normalize our asset paths today. I think that means if you load `"asset.txt"` and `"blah/../asset.txt"`, they are treated as different assets.

Doesn't need to be addressed in this PR, just realized this grossness.